core/optaplanner-core-impl/src/main/java/org/optaplanner/core/config/ruin/RuinPhaseConfig.java

```diff
@@ -0,0 +1,109 @@
+package org.optaplanner.core.config.ruin;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElementWrapper;
+import javax.xml.bind.annotation.XmlTransient;
+import javax.xml.bind.annotation.XmlType;
+
+import org.optaplanner.core.config.heuristic.selector.entity.EntitySelectorConfig;
+import org.optaplanner.core.config.heuristic.selector.move.generic.*;
+import org.optaplanner.core.config.phase.PhaseConfig;
+import org.optaplanner.core.config.solver.termination.TerminationConfig;
+import org.optaplanner.core.config.util.ConfigUtils;
+
+@XmlType(propOrder = {
+        "percentageToRuin",
+        "entitySelectorConfig",
+        "secondaryEntitySelectorConfig",
+        "variableNameIncludeList"
+})
+public class RuinPhaseConfig extends PhaseConfig<RuinPhaseConfig> {
+
+    public static final String XML_ELEMENT_NAME = "ruin";
+
+    @XmlTransient
+    private RuinMoveSelectorConfig moveSelectorConfig = null;
+
+    @XmlElement(name = "entitySelector")
+    private EntitySelectorConfig entitySelectorConfig = null;
+    @XmlElement(name = "secondaryEntitySelector")
+    private EntitySelectorConfig secondaryEntitySelectorConfig = null;
+
+    @XmlElementWrapper(name = "variableNameIncludes")
+    @XmlElement(name = "variableNameInclude")
+    private List<String> variableNameIncludeList = null;
+
+    @XmlElement(name = "percentageToRuin")
+    private Integer percentageToRuin = 20;
```

Comment:
Usually when a magic number appears, it's good to add a comment explaining how you arrived at the number. Providing some context is always a good thing, for the people in the future who have to understand your intentions.

Reply:
For now I left both a comment and todo. I will come back to it when I have the benchmarks